isisd/isis_nb_config.c

```diff
@@ -1843,18 +1843,20 @@ int isis_instance_mpls_ldp_sync_create(struct nb_cb_create_args *args)
 	struct listnode *node;
 	struct isis_circuit *circuit;
 	struct interface *ifp;
-	struct vrf *vrf = vrf_lookup_by_id(VRF_DEFAULT);
-	struct isis *isis = isis_lookup_by_vrfid(VRF_DEFAULT);
+	struct vrf *vrf;
+	struct isis *isis;
+
 
 	switch (args->event) {
 	case NB_EV_VALIDATE:
-		if (isis == NULL)
-			return NB_ERR_VALIDATION;
-		break;
 	case NB_EV_PREPARE:
 	case NB_EV_ABORT:
 		break;
 	case NB_EV_APPLY:
+		vrf = vrf_lookup_by_id(VRF_DEFAULT);
+		isis = isis_lookup_by_vrfid(vrf->vrf_id);
```

Comment:
Please use `area = nb_running_get_entry(args->dnode, NULL, true);` instead as that's the standard way to obtain configuration objects pointers in the NB callbacks (then you can use `area->isis`).

Also, it might be worth adding a check preventing LDP-IGP sync from being configured on IS-IS instances running on a non-default VRF.